Medical uplink, attempt two (Minus conflicts)

Kell-E opened the pull request:
The medical uplink had a large number of items that had little to no realistic use cases and lacked many of the important tools required to be able to actually perform the role of medical technician while playing as part of a group.

The goal was to squish the number of available items but increase the usability of the ones that remained.

First in the list of changes I've added a single use injector filled with hyperzine (3u), synaptizine (1u) and inaprovaline (10u). The injector doesn't facilitate pain immunity alone and instead serves to enhance an antagonists options when it comes to fight or flight, but by no means will it provide them with the ability to ignore incoming fire.

Next up I replaced the stabilization kit with an upgraded slimline version, the slimline can fit into a pocket and takes up much less room in a bag.
The slimline stabilization kit also has an upgraded selection of tools with a focus on allowing an unskilled character to perform emergency first aid in any number of potential situations. It contains

7 autoinjectors
1x Coagulant
1x Tramadol
1x Adrenaline
2x Inaprovaline
1x Dexalin Plus
1x Dylovene

Cost: 16 TC (Unchanged)
*The coagulant injector contains 1u of coagulant and 4u of nanoblood, the coagulant will have a reasonably high chance to heal any internal and external bleeding. Coagulant has no method of creation outside of appearing inside of these autoinjectors.

Lastly I removed the health HUD and scanner from the uplink, the individual items cost more than they really should have been valued at and largely bloated the list of objects.

In their place I've added a new "Advanced medical toolkit" which contains the following

1x Roller bed
1s Box of syringes
1x Health HUD
1x Health Scanner
1x Auto-resuscitator
1x Auto-compressor
1x Nanoblood IV Bag
1x Advanced first-aid kit
1x Nitrile Gloves
Cost: 24 TC
Overall my goal was to improve the quality of available tools without merely making them more powerful, providing options to antagonists so that they can fulfill various roles without resorting to stealing to obtain some of the more basic (and often required) tools.

🆑 Kell-E
rscadd: Added a stimulant autoinjector to the uplink (Cost: 14)
rscadd: Added coagulant (Cannot be created at this time, but does appear in coagulant autoinjectors). Coagulant can staunch internal and external bleeding.
balance: Improved the stabilization kit available in the uplink, it now takes up much less space, fits into a pocket and has had the quality of it's contents improved. Includes one coagulant autoinjector.
rscdel: Removed the medical HUD from the uplink
rscdel: Removed the medical scanner from the uplink
rscadd: Added an advanced medical toolkit to the uplink, contains a large number of medical tools including a HUD, scanner, rollerbed, auto-resuscitator, auto-compressor, nanoblood IV bag, advanced first-aid kit, nitrile gloves and a box of syringes. (Cost: 24)
/🆑

SierraKomodo commented on code/datums/uplink/medical.dm:
```diff
@@ -10,11 +10,17 @@
-	name = "Stabilisation First Aid Kit"
-	desc = "Contains variety of emergency medical pouches."
+	name = "Slimline Stabilisation Kit"
+	desc = "A pocket sized medikit filled with lifesaving equipment."
```
```suggestion
	desc = "A pocket-sized medkit filled with lifesaving equipment."
```

SierraKomodo commented on code/datums/uplink/medical.dm:
```diff
@@ -10,11 +10,17 @@
 
+/datum/uplink_item/item/medical/combatstim
+	name = "Combat Stimulants"
+	desc = "A single use medical injector filled with performance enhancing drugs."
```
```suggestion
	desc = "A single-use medical injector filled with performance enhancing drugs."
```

SierraKomodo commented on code/datums/uplink/medical.dm:
```diff
@@ -28,6 +34,12 @@
 
+/datum/uplink_item/item/medical/advancedmedibag
+	name = "Advanced medical toolkit"
+	desc = "A duffle bag containing a roller bed, syringes, health analyzer, health HUD, auto-compressor, auto-ressuscitator, nanoblood, an advanced fist-aid kit and a pair of nitrile gloves."
```
```suggestion
	desc = "A duffle bag containing a roller bed, syringes, health analyzer, health HUD, auto-compressor, auto-resuscitator, nanoblood, an advanced first-aid kit, and a pair of nitrile gloves."
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -1021,3 +1021,24 @@
+	color = "#bf0000"
+	metabolism = REM * 0.05
+
+/datum/reagent/coagulant/affect_blood(var/mob/living/carbon/M, alien, removed)
```
```suggestion
/datum/reagent/coagulant/affect_blood(mob/living/carbon/M, alien, removed)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -1021,3 +1021,24 @@
+				if(W.bleeding() && prob(20))
+					W.bleed_timer = 0
+					W.clamped = TRUE
+					E.status &= ~ORGAN_BLEEDING
```
What if there's more than one bleeding wound and the prob check doesn't close all of them?

Kell-E replied:
The odds are overwhelmingly in favor of it closing every single bleeding wound you might have over it's duration, but I didn't want people to siphon it out and use it in 0.1 shots to instantly close all wounds on demand.

SierraKomodo replied:
The `E.status &= ~ORGAN_BLEEDING` call might result in stopping all bleeding as far as mechanics are concerned, even if only one wound is closed. The underlying code for this is a bit of a mess, though. This will probably have to be another 'Look at this later' item.

Kell-E replied:
I just ran with whatever worked, I thought that it'd only work on the injury it was called on but I can't recall if I checked that when I tested it out.

SierraKomodo replied:
To clarify, 'stopping all bleeding on the specific organ, not the whole mob.' But yeah, I'll tag this one as 'Look at it later.' There are other things in code doing this same thing, it looks like, so it's not new.

SierraKomodo reviewed (changes requested):
The coagulant is coming up as an unknown substance on scans. This should probably be a known thing for antag medic purposes. This should be fixable by setting `scannable = TRUE` in the base definition for the reagent.

Kell-E commented:
Whoops, lemme fix that real quick.

Kell-E committed:
Medical uplink, attempt two (Minus conflicts)

The medical uplink had a large number of items that had little to no realistic use cases and lacked many of the important tools required to be able to actually perform the role of medical technician while playing as part of a group.

The goal was to squish the number of available items but increase the usability of the ones that remained.

First in the list of changes I've added a single use injector filled with hyperzine (3u), synaptizine (1u) and inaprovaline (10u). The injector doesn't facilitate pain immunity alone and instead serves to enhance an antagonists options when it comes to fight or flight, but by no means will it provide them with the ability to ignore incoming fire.

Next up I replaced the stabilization kit with an upgraded slimline version, the slimline can fit into a pocket and takes up much less room in a bag.
The slimline stabilization kit also has an upgraded selection of tools with a focus on allowing an unskilled character to perform emergency first aid in any number of potential situations. It contains

7 autoinjectors
1x Coagulant
1x Tramadol
1x Adrenaline
2x Inaprovaline
1x Dexalin Plus
1x Dylovene

Cost: 16 TC (Unchanged)
*The coagulant injector contains 1u of coagulant and 4u of nanoblood, the coagulant will have a reasonably high chance to heal any internal and external bleeding. Coagulant has no method of creation outside of appearing inside of these autoinjectors.

Lastly I removed the health HUD and scanner from the uplink, the individual items cost more than they really should have been valued at and largely bloated the list of objects.

In their place I've added a new "Advanced medical toolkit" which contains the following

1x Roller bed
1s Box of syringes
1x Health HUD
1x Health Scanner
1x Auto-resuscitator
1x Auto-compressor
1x Nanoblood IV Bag
1x Advanced first-aid kit
1x Nitrile Gloves
Cost: 24 TC
Overall my goal was to improve the quality of available tools without merely making them more powerful, providing options to antagonists so that they can fulfill various roles without resorting to stealing to obtain some of the more basic (and often required) tools.

🆑 Kell-E
rscadd: Added a stimulant autoinjector to the uplink (Cost: 14)
rscadd: Added coagulant (Cannot be created at this time, but does appear in coagulant autoinjectors). Coagulant can staunch internal and external bleeding.
balance: Improved the stabilization kit available in the uplink, it now takes up much less space, fits into a pocket and has had the quality of it's contents improved. Includes one coagulant autoinjector.
rscdel: Removed the medical HUD from the uplink
rscdel: Removed the medical scanner from the uplink
rscadd: Added an advanced medical toolkit to the uplink, contains a large number of medical tools including a HUD, scanner, rollerbed, auto-resuscitator, auto-compressor, nanoblood IV bag, advanced first-aid kit, nitrile gloves and a box of syringes. (Cost: 24)
/🆑
Changes based upon review


scan

a
